<!-- Launch File for running svo with ceres backend from bag-file -->
<!-- Xie Chen, 2022/01/13 -->
<launch>

  <!-- calibration params: use stereo file to simulate 2 cameras -->  
  <arg name="calib_file" default="$(find svo_ros)/param/calib/euroc_stereo.yaml"/>

  <!-- SVO node -->
  <node pkg="svo_ros" type="svo_node" name="svo" clear_params="true" output="screen" args="--v=0" >

    <!-- Suppose we have 2 cameras, because SVO treats them independently -->
    <param name="cam0_topic" value="/cam0/image_raw" type="str" />
    <param name="cam1_topic" value="/cam1/image_raw" type="str" />
	
    <!-- Imu topic to subscribe to -->
    <param name="imu_topic" value="/imu0" type="str" />

    <!-- Camera calibration file -->
    <param name="calib_file" value="$(arg calib_file)" />

    <!--Parameters-->
    <rosparam file="$(find svo_ros)/param/vio_mono.yaml" />
    <param name="runlc" value="true" /> <!--Loop Closing-->
    <param name="multi_cam" value="true" />  <!--Whether to use multi-cameras-->

  </node>

  <!-- RVIZ default config: rviz_config_vio.rviz  -->
  <node name="vis" pkg="rviz" type="rviz" args=" -d $(find svo_ros)/rviz_config_mono_2.rviz" />

</launch>
